### PR TITLE
feat: use scatterplot selections to create tags

### PR DESCRIPTION
**Pull request by joshjzhou**

### CODE REVIEW

File: frontend/src/metadata/MetadataPanel.svelte [hunk 1]
```diff
@@ -285,6 +289,43 @@
 		<SliceCell slice={s} />
 	{/each}
 
+	<div id="tag-header" class="inline" style:margin-top="10px">
+		<div class="inline">
+			<h4>Metadata Tags</h4>
```
Comment (cabreraalex) [inline frontend/src/metadata/MetadataPanel.svelte:294]: Just "Tags" is fine

File: zeno/classes/tag.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+from zeno.classes.base import CamelModel
+
+class Tag(CamelModel):
+    tag_name: str
+    folder: str
+    selection_ids: List[str]
+
+
+class TagMetric(CamelModel):
```
Comment (cabreraalex) [inline zeno/classes/tag.py:11]: Maybe we create just one "MetricResult" object that both slice and tag use?

File: zeno/backend.py [hunk 3]
```diff
@@ -65,6 +66,10 @@ def __init__(self, params: ZenoParameters):
             "slices.pickle", self.cache_path, {}
         )
 
+        self.tags: Dict[str, Tag] = read_pickle(
```
Comment (cabreraalex) [inline zeno/backend.py:69]: remove newline above

File: frontend/src/App.svelte [hunk 4]
```diff
@@ -42,6 +43,7 @@
 		models.set(inits.models);
 		metrics.set(inits.metrics);
 		folders.set(inits.folders);
+		// tags.set(inits.tags);
```
Comment (cabreraalex) [inline frontend/src/App.svelte:46]: remove

**Review by cabreraalex — COMMENTED** [review]
Just a few small comments, it's looking great!

File: zeno/classes/classes.py [hunk 5]
```diff
@@ -48,9 +49,15 @@ class TableRequest(CamelModel):
     slice_range: List[int]
     filter_predicates: FilterPredicateGroup
     sort: Tuple[Union[ZenoColumn, None], bool]
+    tag_ids: FilterIds
     filter_ids: Optional[FilterIds] = None
 
 
+class PlotRequest(CamelModel):
```
Comment (cabreraalex) [inline zeno/classes/classes.py:56]: What is this used for? IDK if I like the name too much. Any other options?

File: frontend/src/metadata/MetadataPanel.svelte [hunk 6]
```diff
@@ -306,6 +376,62 @@
 		<SliceCell slice={s} />
 	{/each}
 
+	<div id="tag-header" class="inline" style:margin-top="10px">
+		<div class="inline">
```
Comment (cabreraalex) [inline frontend/src/metadata/MetadataPanel.svelte:380]: I also can't decide how the sticky headers should work. If you scroll on the left you'll see the slice and metadata headers stick to the top. I think the tag one should too? But then it gets a bit busy... maybe we can ask erica

File: frontend/src/theme/_smui-theme.scss [hunk 7]
```diff
@@ -24,6 +24,7 @@
 	--P4: #f9f7fb;
 	--Y1: #f2f2ee;
 	--Y2: #fbfbfa;
+	--N2: #d0ebe6;
```
Comment (cabreraalex) [inline frontend/src/theme/_smui-theme.scss:27]: What do you think about a slightly lighter green like #e3f8f4

**Review by cabreraalex — CHANGES_REQUESTED** [review]
Overall it looks great! I'm super impressed at how you managed to understand the whole histogram API interaction and integrate the tags into updating the table requests across the codebase. Very very cool.

File: zeno/classes/classes.py (hunk 5 above)
Comment (joshjzhou) [reply zeno/classes/classes.py:56]: This is used to query all the IDs that have been selected by filter predicates and tags for the scatterplot (getFilteredIds function), which is why I named it PlotRequest to mimic the TableRequest naming.

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**